Swift/Tests/DictionaryTest.swift

```diff
@@ -75,6 +77,47 @@ class DictionaryTest: CBLTestCase {
         })
     }
     
+    func testSettersDictionary() throws {
+        let dict = MutableDictionaryObject()
+        dict.setValue("value", forKey: "key")
+        XCTAssertNotNil(dict.value(forKey: "key"))
+        dict.setValue(0, forKey: "key")
+        
+        dict.setString("string", forKey: "key")
+        XCTAssertNotNil(dict.string(forKey: "key"))
+        
+        dict.setNumber(1, forKey: "key")
+        XCTAssertNotNil(dict.number(forKey: "key"))
+        
+        dict.setInt(0, forKey: "key")
```

Comment:
I would just non zero number as the getter will return zero for non integer as well.